pkg/ocm/accounts_mgmt.go

```diff
@@ -81,30 +83,44 @@ func (a accountsMgmt) GetSubscription(ctx context.Context, subscriptionID strfmt
 	return responseVal, nil
 }
 
+func (a accountsMgmt) updateSubscription(ctx context.Context, subscriptionID strfmt.UUID, sub *amgmtv1.Subscription, err error) error {
+	if err != nil {
+		a.client.logger.Error(ctx, "Failed to create subscription request. Error: %v", err)
+		return err
+	}
+	return a.sendSubscriptionUpdateRequest(ctx, subscriptionID, sub)
+}
+
 // This function updates all the subscription's data that we didn't know at the subscription creation time
 // and we know only after installation is done such openshift_cluster_id.
 func (a accountsMgmt) UpdateSubscriptionPostInstallation(ctx context.Context, subscriptionID, openshiftClusterID strfmt.UUID) error {
 	defer commonutils.MeasureOperation("OCM-UpdateSubscriptionPostInstallation", a.client.log, a.client.metricsApi)()
 
 	sub, err := amgmtv1.NewSubscription().ExternalClusterID(openshiftClusterID.String()).Status(SubscriptionStatusActive).Build()
-	if err != nil {
-		a.client.logger.Error(ctx, "Failed to create subscription request. Error: %v", err)
-		return err
-	}
-
-	return a.sendSubscriptionUpdateRequest(ctx, subscriptionID, sub)
+	return a.updateSubscription(ctx, subscriptionID, sub, err)
 }
 
 func (a accountsMgmt) UpdateSubscriptionDisplayName(ctx context.Context, subscriptionID strfmt.UUID, displayName string) error {
 	defer commonutils.MeasureOperation("OCM-UpdateSubscriptionDisplayName", a.client.log, a.client.metricsApi)()
 
 	sub, err := amgmtv1.NewSubscription().DisplayName(displayName).Build()
-	if err != nil {
-		a.client.logger.Error(ctx, "Failed to create subscription request. Error: %v", err)
-		return err
+	return a.updateSubscription(ctx, subscriptionID, sub, err)
+}
+
+func (a accountsMgmt) UpdateSubscriptionConsoleUrlIfNeeded(ctx context.Context, subscriptionID strfmt.UUID, consoleUrl string) error {
+
+	if a.isConsoleUrlSet {
+		return nil
 	}
+	defer commonutils.MeasureOperation("OCM-UpdateSubscriptionConsoleUrl", a.client.log, a.client.metricsApi)()
```

Comment:
Isn't that static? It would affect all the clusters

Reply:
Not sure I understand your comment.

Reply:
Maybe I don't understand this module correctly. But isn't `accountsMgmt` struct is not cluster-dependant? 
From the clusterAPI it is being called as `m.ocmClient.AccountsMgmt`, hence - every call, regardless of the cluster - would have the same `isConsoleUrlSet` variable, which should be cluster-dependant.

Reply:
Ohh I see what you mean, this is a good point, I will fix it and make this status variable be a `map[strfmt.UUID]bool`.